autorepro/utils/error_handling.py

```diff
@@ -0,0 +1,431 @@
+"""
+Standardized error handling utilities for subprocess and file operations.
+
+This module provides consistent error handling patterns to replace scattered
+exception handling across the AutoRepro codebase, particularly for subprocess
+execution and file operations.
+"""
+
+import contextlib
+import logging
+import subprocess
+from collections.abc import Generator
+from dataclasses import dataclass
+from pathlib import Path
+
+from .file_ops import FileOperations
+from .process import SubprocessConfig
+
+
+@dataclass
+class ErrorContext:
+    """Context information for error reporting."""
+    operation: str | None = None
+    path: Path | str | None = None
+    log_operations: bool = False
+
+
+class AutoReproError(Exception):
+    """Base exception class for AutoRepro operations."""
+
+    def __init__(self, message: str, operation: str | None = None, cause: Exception | None = None):
+        super().__init__(message)
+        self.message = message
+        self.operation = operation
+        self.cause = cause
+
+
+class SubprocessError(AutoReproError):
+    """Standardized exception for subprocess execution errors."""
+
+    def __init__(
```

Comment:
Solution: Refactor the __init__ method to accept fewer arguments.
Consider grouping related arguments into a class or dictionary, or refactor logic to reduce argument count.